ui/tests/unit/services/api-test.js

```diff
@@ -0,0 +1,202 @@
+    assert.true(this.tokenForUrl.calledWith(context.url), 'Url is passed to tokenForUrl method');
+    assert.deepEqual(context, noTokenContext, 'Original context is returned when no token is present');
+
+    this.tokenForUrl.returns(this.wrapInfo);
+
+    const { token } = this.wrapInfo;
+    const tokenContext = await this.apiService.getControlGroupToken(context);
+    const newContext = {
+      url: '/v1/sys/wrapping/unwrap',
+      init: {
+        method: 'POST',
+        headers: { 'X-Vault-Token': token },
+        body: JSON.stringify({ token }),
+      },
+    };
+
+    assert.deepEqual(tokenContext, newContext, 'New context is returned when token is present');
+  });
+
+  test('it should set headers', async function (assert) {
+    const headers = {
+      'X-Vault-Token': 'root',
+      'X-Vault-Namespace': 'ns1',
+    };
+
+    const {
+      init: { headers: originalHeaders },
+    } = await this.apiService.setHeaders({ init: { headers } });
+
+    assert.strictEqual(
+      originalHeaders.get('X-Vault-Token'),
+      'root',
+      'Original token header value is preserved'
+    );
+    assert.strictEqual(
+      originalHeaders.get('X-Vault-Namespace'),
+      'ns1',
+      'Original namespace header value is preserved'
+    );
+
+    const {
+      init: { headers: newHeaders },
+    } = await this.apiService.setHeaders({ init: {} });
+
+    assert.strictEqual(
+      newHeaders.get('X-Vault-Token'),
+      'foobar',
+      'New token header value from auth service is set'
+    );
+    assert.strictEqual(
+      newHeaders.get('X-Vault-Namespace'),
+      'another-ns',
+      'New namespace header value from namespace service is set'
+    );
+
+    const {
+      init: { headers: contentTypeHeaders },
+    } = await this.apiService.setHeaders({ init: { method: 'PATCH' } });
+
+    assert.strictEqual(
+      contentTypeHeaders.get('Content-Type'),
+      'application/merge-patch+json',
+      'Content type header is set for PATCH method'
+    );
+  });
+
+  test('it should show warnings', async function (assert) {
+    const warnings = ['warning1', 'warning2'];
+    const response = new Response(JSON.stringify({ warnings }));
+
+    await this.apiService.showWarnings({ response });
+
+    assert.true(this.info.firstCall.calledWith(warnings[0]), 'First warning message is shown');
+    assert.true(this.info.secondCall.calledWith(warnings[1]), 'Second warning message is shown');
+
+    this.info.resetHistory();
+    await this.apiService.showWarnings({ response, url: '/v1/sys/internal/counters/activity' });
+
+    assert.true(this.info.notCalled, 'Warnings are suppressed for activity endpoint');
+  });
+
+  test('it should delete control group token', async function (assert) {
+    await this.apiService.deleteControlGroupToken({ url: this.url });
+
+    assert.true(this.tokenForUrl.calledWith(this.url), 'Url is passed to tokenForUrl method');
+    assert.true(
+      this.deleteControlGroupToken.calledWith(this.wrapInfo.accessor),
```

Comment:
why is this called with the accessor?

Reply:
This was [ported over from the application adapter](https://github.com/hashicorp/vault/blob/main/ui/app/adapters/application.js#L90-L92). This is just basically asserting that the call is made with correct arg to remove the token. In the [control group service method](https://github.com/hashicorp/vault/blob/main/ui/app/services/control-group.js#L50-L54) it is using `keyFromAccessor(accessor)` to get the key that the token was stored under in local storage and removing it.